openpype/hosts/nuke/api/plugin.py

```diff
@@ -808,6 +817,18 @@ def __init__(self,
         self.log.info("File info was set...")
 
         self.file = self.fhead + self.name + ".{}".format(self.ext)
+        if ".{}".format(self.ext) not in VIDEO_EXTENSIONS:
+            # filename would be with frame hashes if
+            # the file extension is not in video format
+            filename = get_head_filename_without_hashes(
+                self.path_in, self.name)
+            self.file = filename
+            # make sure the filename are in
+            # correct image output format
+            if ".{}".format(self.ext) not in self.file:
+                original_ext = os.path.splitext(filename)[-1].strip(".")    # noqa
+                self.file = filename.replace(original_ext, ext)
```

Comment:
Looking at this logic we're much better off just always setting the output extension we want instead of replacing `original_ext` to `ext` in the filename, etc. even though what we really want is to ever just swap around the extension - so let's just do that.
```suggestion
            # ensure output has expected extension
            filename_no_ext, _ = os.path.splitext(filename)
            self.file = "{}.{}".format(filename_no_ext, self.ext)
```
No need for the if statement and much clearer that we're always enforcing the extension `self.ext`